packages/integrations/markdoc/src/extensions/shiki.ts

```diff
@@ -0,0 +1,126 @@
+// @ts-expect-error Cannot find module 'astro/runtime/server/index.js' or its corresponding type declarations.
+import { unescapeHTML } from 'astro/runtime/server/index.js';
+import type { ShikiConfig } from 'astro';
+import type * as shikiTypes from 'shiki';
+import type { AstroMarkdocConfig } from '../config.js';
+import Markdoc from '@markdoc/markdoc';
+import { getHighlighter } from 'shiki';
+
+// Map of old theme names to new names to preserve compatibility when we upgrade shiki
+const compatThemes: Record<string, string> = {
+	'material-darker': 'material-theme-darker',
+	'material-default': 'material-theme',
+	'material-lighter': 'material-theme-lighter',
+	'material-ocean': 'material-theme-ocean',
+	'material-palenight': 'material-theme-palenight',
+};
+
+const normalizeTheme = (theme: string | shikiTypes.IShikiTheme) => {
+	if (typeof theme === 'string') {
+		return compatThemes[theme] || theme;
+	} else if (compatThemes[theme.name]) {
+		return { ...theme, name: compatThemes[theme.name] };
+	} else {
+		return theme;
+	}
+};
+
+/**
+ * Cache each Shiki highlighter by theme name.
+ * Note: cache only needed for dev server reloads, internal test suites, and manual calls to `Markdoc.transform` by the user.
+ * Otherwise, `shiki()` is only called once per build, NOT once per page, so a cache isn't needed!
+ */
+const highlighterCacheAsync = new Map<string, Promise<shikiTypes.Highlighter>>();
```

Comment:
It's really weird caching the **promises**. Shouldn't we cache the actual values?

Reply:
I took this from our existing implementation and it took me a bit to see _why._ In short: this should cache promises for multiple parallel requests with lower likelihood of cache misses. This benefit is lower because of my comment there, so I'd be open to refactoring

Reply:
Yeah we have a few shiki code copy-pasted around the codebase. We should probably consolidate it as part of the markdown packages in the future.